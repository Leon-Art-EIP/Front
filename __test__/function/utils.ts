import type { ReactTestRendererJSON } from "react-test-renderer";
// eslint-disable-next-line import/no-extraneous-dependencies
import renderer from "react-test-renderer";
import { expect } from "vitest";

function toJson(component: renderer.ReactTestRenderer): ReactTestRendererJSON {
  const result = component.toJSON();
  expect(result).toBeDefined();

  return result as ReactTestRendererJSON;
}

export function testSnapshot(element: JSX.Element): void {
  const component = renderer.create(element);
  const tree = toJson(component);
  expect(tree).toMatchSnapshot();
}
