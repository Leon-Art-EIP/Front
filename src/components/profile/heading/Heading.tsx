import Image from "next/image";
import ProfilePicture from "../profilePicture/ProfilePicture";

interface IHeadingProps {
  profilePicture: string;
  banner: string;
}

export default function Heading(props: IHeadingProps): JSX.Element {
  return (
    <div className="grid grid-cols-4 relative h-64">
      <Image src={props.banner} alt="profileBanner" className="absolute h-64 z-0" />
      <div></div>
      <div></div>
      <div></div>
      <div className="z-10 h-full flex items-center justify-center p-5">
        <ProfilePicture src={props.profilePicture} width={200} height={200} />
      </div>
    </div>
  );
}
