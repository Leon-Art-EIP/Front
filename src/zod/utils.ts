export const getValueOrUndefined = (arg: unknown) =>
  arg === "" || arg === undefined || arg === null ? undefined : arg;
