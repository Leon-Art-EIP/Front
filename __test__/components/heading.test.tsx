import { test } from "vitest";
import { testSnapshot } from "../function/utils";
import Heading from "../../src/components/profile/heading/Heading";
import leonart from "../../public/leonart_logo.png";

test("Heading snapshot", () => {
  testSnapshot(<Heading banner={leonart} profilePicture={leonart} />);
});
