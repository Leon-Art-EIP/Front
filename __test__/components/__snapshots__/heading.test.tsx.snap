// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Heading snapshot 1`] = `
<div
  className="shrink-0 relative h-64"
>
  <div
    className="z-10 flex items-center lg:justify-end justify-center lg:mx-28 h-full"
  >
    <img
      alt="profilePicture"
      className="rounded-full border-4 z-10 w-48 h-48 object-contain bg-black"
      height={192}
      src="/public/leonart_logo.png"
      width={192}
    />
  </div>
  <img
    alt="profileBanner"
    className="absolute inset-0 h-64 z-0 w-screen object-contain bg-secondary"
    height={256}
    src="/public/leonart_logo.png"
  />
</div>
`;
